server/ai_mediaserver.go

```diff
@@ -900,16 +901,16 @@ func (ls *LivepeerServer) WithCode(code int) http.Handler {
 	})
 }
 
-func (ls *LivepeerServer) cleanupLive(ctx context.Context, stream string) {
+func cleanupLive(ctx context.Context, node *core.LivepeerNode, stream string) {
 	clog.Infof(ctx, "Live video pipeline finished")
-	ls.LivepeerNode.LiveMu.Lock()
-	pub, ok := ls.LivepeerNode.LivePipelines[stream]
-	delete(ls.LivepeerNode.LivePipelines, stream)
+	node.LiveMu.Lock()
+	pub, ok := node.LivePipelines[stream]
+	delete(node.LivePipelines, stream)
```

Comment:
Since this is called from multiple places now, wdyt of having an explicit logic like this here?
```suggestion
	node.LiveMu.Lock()
	pub, ok := node.LivePipelines[stream]
	if !ok {
	    // already cleaned up
	    node.LiveMy.Unlock()
	    return
	}
	delete(node.LivePipelines, stream)
```
the no need to check `ok` below either